packages/wonder-blocks-dropdown/components/search-text-input.js

```diff
@@ -0,0 +1,133 @@
+import IconButton from "@khanacademy/wonder-blocks-icon-button";
+import Icon, {icons} from "@khanacademy/wonder-blocks-icon";
+import Color from "@khanacademy/wonder-blocks-color";
+import Spacing from "@khanacademy/wonder-blocks-spacing";
+import type {StyleType} from "@khanacademy/wonder-blocks-core";
+
+type Props = {|
+    searchText: string,
+    onChange: (searchText: string) => mixed,
+    onClick: () => mixed,
+    itemRef: {current: any},
+    style?: StyleType,
+|};
+
+type State = {|
+    focused: boolean,
+|};
+
+export default class SearchTextInput extends React.Component<Props, State> {
+    state = {
+        focused: false,
+    };
+
+    handleChange = (e: SyntheticInputEvent<>) => {
+        e.preventDefault();
+        this.props.onChange(e.target.value);
+    };
+
+    handleDismiss = () => {
+        const {onClick, onChange} = this.props;
+        // Empty the search text and focus the SearchTextInput
+        onChange("");
+        onClick();
+    };
+
+    handleBlur = (e: SyntheticInputEvent<>) => {
+        this.setState({focused: false});
+    };
+
+    handleFocus = (e: SyntheticInputEvent<>) => {
+        this.setState({focused: true});
+    };
+
+    render() {
+        const {onClick, itemRef, searchText, style} = this.props;
+        // TODO(jangmi): Use translated strings for "Filter", "Clear search"
+        return (
+            <View
+                onClick={onClick}
+                style={[
+                    styles.inputContainer,
+                    this.state.focused && styles.focused,
+                    style,
+                ]}
+            >
+                <Icon
+                    icon={icons.search}
+                    size="medium"
+                    color={Color.offBlack50}
+                    style={styles.searchIcon}
+                    aria-hidden="true"
+                />
+                <input
+                    type="text"
+                    onChange={this.handleChange}
+                    onFocus={this.handleFocus}
+                    onBlur={this.handleBlur}
+                    ref={itemRef}
+                    placeholder="Filter"
+                    value={searchText}
+                    className={css(
+                        styles.inputStyleReset,
+                        typographyStyles.LabelMedium,
+                    )}
+                />
+                {searchText.length > 0 && (
+                    <IconButton
+                        icon={icons.dismiss}
+                        kind="tertiary"
+                        onClick={this.handleDismiss}
+                        style={styles.dismissIcon}
+                        aria-label="Clear search"
+                    />
+                )}
+            </View>
+        );
+    }
+}
+
+const styles = StyleSheet.create({
+    inputContainer: {
+        flexDirection: "row",
+        border: `1px solid ${Color.offBlack16}`,
+        borderRadius: 4,
```

Comment:
Perhaps use a Wonder Blocks Spacing constant?

Reply:
Hmm I generally don't use WB Spacing for borderRadius because border radius 4px means different than 4px spacing. Is that recommended use case for WB Spacing?